packages/good-design/src/core/layout/Trans.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { Trans as LinguiTrans } from "@lingui/react";
+
+import { IButtonProps, ITextProps, Text } from "native-base";
+import Title from "./Title";
+import { GoodButton } from "../buttons";
+
+/**
+ * Use this component when wanting to apply Text variants (see theme.tsx)
+ * @param t text that needs to supported translation
+ * @returns Text
+ */
+export const TransText = ({ t, ...props }: { t: string } & ITextProps) => (
```

Comment:
i dont understand why we need this helpers. it is much more verbose that developer simply has to write
`<Button><Trans>text</Trans></Button>` and so on

Reply:
So that I can apply styling props directly to the trans object.

In your example with the button, you wont use this helpers but do it as you show.

Reply:
i still dont understand. why do we need this complication with render=...
give me an example unlike my button example where you would need to use this helper

Reply:
It's not a hard 'need'.

I just compared these three options:
```
<TransText height={20} width={20} px={2} t={*/i18n/* `Some text to translate`} />

<Text height={20} width={20} px={2}>
  <Trans component={null}>
    {*/i18n/* `Some text to translate`}
  </Trans>
</Text>


<Trans render={({ translation }) => <Text label={translation} />}>{*/i18n/* "Sign in"}</Trans>;
```

And I preferred the first one to not over-nest components, have complicated inline definitions.